cohorts/cohort.py

```diff
@@ -1014,15 +1034,8 @@ def plot_boolean(self,
         (Test statistic, p-value): (float, float)
 
         """
-        cols, df = self.as_dataframe(on, col, **kwargs)
-        if type(cols) == str:
-            if plot_col is not None:
-                raise ValueError("plot_col is specified when it isn't ndeeded.")
-            plot_col = cols
-        elif type(cols) == list:
-            if plot_col is None:
-                raise ValueError("plot_col must be specified when multiple `on`s are present.")
-
+        cols, df = self.as_dataframe(on, return_cols=True, **kwargs)
+        plot_col = self.plot_col_from_cols(cols=cols, plot_col=plot_col)
```

Comment:
Good question: we might want to use `on` to grab not just e.g. `snv_count` in `cohort.plot_benefit(on=snv_count)`, but actually to grab the `benefit_col` itself; sometimes we want to split patients by a column that isn't already in the `DataFrame`. I'll mention this here?